Tickr/Tickr/StocksTableViewController.swift

```diff
@@ -0,0 +1,355 @@
+
+import UIKit
+
+class StocksTableViewController: UIViewController {
+    //Class Variables
+    lazy var notificationCenter: NSNotificationCenter = {
+        return NSNotificationCenter.defaultCenter()
+    }()
+    
+    //Outlets
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    
+    //MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchBar.delegate = self
+        self.fetchStockUpdates()
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        //Listen for any updates from the Stock Manager
+        notificationCenter.addObserver(
+            self,
+            selector: #selector(StocksTableViewController.stocksWereUpdated(_:)),
+            name: Constants.kNotificationStockPricesUpdated,
+            object: nil
+        )
+        
+        //Listen for any stock price alerts from the WatchList Manager
+        notificationCenter.addObserver(
+            self,
+            selector: #selector(StocksTableViewController.priceTargetWasHit(_:)),
+            name: Constants.kPriceTargetWasHit,
+            object: nil
+        )
+        
+        //Reload tableView to reflect any changes in the search view
+        tableView.reloadData()
+    }
+    
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        //No reason to update the stock data when the stock view isn't visible
+        notificationCenter.removeObserver(self)
+    }
+
+    
+    //MARK: - Stock Updates
+    /*
+        This will fetch current stock prices on a set interval.
+        This interval can be changed in Constants.swift.
+     
+        Stocks are added to the WatchList stock array. StockManager then
+        makes a query on the stocks in that array. Those stock updates 
+        are parsed, and then sent back here using a NSNotification. 
+        From there, they're added to the local array and displayed.
+     */
+    func fetchStockUpdates() {
+        
+        //Update any stocks in the WatchList
+        StockManager.sharedInstance.fetchListOfSymbols(WatchListManager.sharedInstance.stocks)
+        
+        //This will call fetchStockUpdates only after the update interval has elapsed
+        dispatch_after(
+            dispatch_time(
+                DISPATCH_TIME_NOW,
+                Int64(Constants.kUpdateInterval * Double(NSEC_PER_SEC))
+            ),
+            dispatch_get_main_queue(),
+            {
+                self.fetchStockUpdates()
+            }
+        )
+    }
+    
+    /*
+        This will be called when a stock update notification is received
+     
+        - parameter notification: a notification containing a userInfo dictionary that 
+                                  contains data on the stocks that were updated.
+     */
+    func stocksWereUpdated(notification: NSNotification) {
```

Comment:
I missed that when refactoring. I was originally passing the stock updates to my VC via NSNotification.userInfo and then realized that my view shouldn't be parsing that and so I removed it but forgot to remove the parameter. Nice find.